openapi/servicenow/client.bal

```diff
@@ -0,0 +1,477 @@
+        return response;
+    }
+    # Deletes the specified record from the specified table.
+    #
+    # + tableName - Name of the table from which to delete the specified record, such as "incident" or "asset".
+    # + sys_id - Sys_id of the record to delete.
+    # + sysparmQueryNoDomain - Flag that indicates whether to restrict the record search to only the domains for which the logged in user is configured.
+    # + accept - Data format of the response body.
+    # + return - **HTTP Status Code:** `204`. Meeting updated.
+    remote isolated function deleteRecordById(string tableName, string sys_id, boolean? sysparmQueryNoDomain = (), string? accept = ()) returns http:Response|error {
+        string  path = string `/api/now/table/${tableName}/${sys_id}`;
+        map<anydata> queryParam = {"sysparm_query_no_domain": sysparmQueryNoDomain};
+        path = path + check getPathForQueryParam(queryParam);
+        map<any> headerValues = {"Accept": accept};
+        map<string|string[]> accHeaders = getMapForHeaders(headerValues);
+        http:Request request = new;
+        //TODO: Update the request as needed;
+        http:Response response = check self.clientEp-> delete(path, request, headers = accHeaders, targetType = http:Response);
+        return response;
+    }
+    # Updates the specified record with the name-value pairs included in the request body.
+    #
+    # + tableName - Name of the table in which to the specified record is located.
+    # + sys_id - Sys_id of the record to update.
+    # + payload - Field name and the associated value for each parameter to define in the specified record in JSON format.
+    # + sysparmDisplayValue - Data retrieval operation for reference and choice fields.
+    # + sysparmFields - Comma-separated list of fields to return in the response.
+    # + sysparmInputDisplayValue - Flag that indicates whether to set field values using the display value or the actual value.
+    # + sysparmQueryNoDomain - Flag that indicates whether to restrict the record search to only the domains for which the logged in user is configured.
+    # + sysparmView - UI view for which to render the data. Determines the fields returned in the response.
+    # + accept - Data format of the response body.
+    # + contentType - Data format of the request body.
+    # + xNoResponseBody - By default, responses include body content detailing the new record. Set this header to true in the request to suppress the response body.
+    # + return - If successful, returns a JSON object that includes updated record of the table. Otherwise returns the relevant error.
+    remote isolated function patchRecordById(string tableName, string sys_id, json payload, string? sysparmDisplayValue = (), string? sysparmFields = (), boolean? sysparmInputDisplayValue = (), boolean? sysparmQueryNoDomain = (), string? sysparmView = (), string? accept = (), string? contentType = (), string? xNoResponseBody = ()) returns json|error {
```

Comment:
What is the difference between update and patch? I wanted to understand the behavior change of these two functions.

Reply:
According to the ServiceNow's official documentation the difference is mentioned as below:  

- PUT(in connector it mentioned as update()) : Updates the specified record with the request body.
- PATCH(in connector it mentioned as patchRecordById()) : Updates the specified record with the name-value pairs included in the request body.